### DISCUSSION TITLE
Subscriptions with redis

### DISCUSSION THREAD

**Discussion by ajmerik121** [enisdenjo/graphql-ws#259] [Q&A] [answered]
Why do I get always get this error when I try to listen to events from redis

Redis library for subscriptions
`graphql-redis-subscriptions`

Error
`Internal error occurred during message handling. Please check your implementation. Error: Subscription field must return Async Iterable. Received: { subscribe: [function subscribe] }.
`

Example schema
``const schema = buildSchema(`
    type Query {
        hello: String
    }
    type Subscription {
        messages: String!
    }
`);``

Example resolvers
`const roots = {
    query: {
        hello: () => 'Hello World!',
    },
    subscription: {
        messages: {
            subscribe : async () => await pubsub.asyncIterator("MESSAGE_ADDED")
        }
    },
};`

**Comment by enisdenjo** [comment, marked as answer]
Hey there, the error messages describes exactly what the issue is, you've read it right?
```diff
const roots = {
  query: { hello: () => 'Hello World!' },
  subscription: {
-   messages: {
-     subscribe: async () => await pubsub.asyncIterator('MESSAGE_ADDED'),
-   },
+   messages: async () => await pubsub.asyncIterator('MESSAGE_ADDED'),
  },
};
```
See also: https://github.com/enisdenjo/graphql-ws/issues/205#issuecomment-862165684.
